docs/isl-2-0/spec.md

```diff
@@ -0,0 +1,762 @@
+Note that `timestamp` values that do not have a time component (that is `YYYYT`, `YYYY-MMT`, and `YYYY-MM-DDT` timestamps) have an unknown offset (`-00:00`).
+Timestamps that have an unknown offset are UTC timestamps that make no assertion about the offset in which they occurred.
+The value `null.timestamp` is never in the bounds of a timestamp range, and `timestamp` ranges do not include values of any other type.
+
+# Open Content
+
+{% include grammar-element.md productions="user_content_declaration,user_content_declaration_field" %}
+
+The Ion Schema 2.0 specification allows users to insert additional content into a schema document that is not part of the Ion Schema Language.
+This additional content may be used for documentation, integrations with other tools or frameworks, or any other desired purpose.
+
+A schema header, type definition, or schema footer may include extra fields that are not explicitly stated in the Ion Schema specification.
+These extra fields may have any field name that is not a *keyword*.
+If the field name is a *reserved symbol*, its use as open content must be declared in the appropriate subfield of the `user_content` field in the schema header.
+For example, the reserved symbol `list_type` may be used in a type definition if `user_content: { type: [ list_type ] }` is present in the schema header.
+
+{% capture sample_code %}
+```ion
+schema_header::{
+  info: "This schema is about penguins.",
+  user_content: {
+    schema_header: [info],
+    type: [region, crested, banded],
+  }
+}
+type::{
+  name: adelie,
+  type: penguin,
+  region: antarctica,
+  crested: false,
+  banded: false,
+}
+type::{
+  name: humboldt,
+  type: penguin,
+  region: south_america,
+  crested: false,
+  banded: true,
+}
+schema_footer::{}
+```
+{% endcapture %}
+{% include example.md title="A schema with open content using reserved symbols as open content field names." markdown=sample_code %}
+
+An Ion Schema MAY include extra top-level values that are not explicitly specified in the Ion Schema specification, but any top-level open content MUST NOT be annotated with a *reserved symbol*.
+Top-level open content may appear before, after, or in between the Ion Schema version marker, header, types, and footer.
+Note that Ion Schema version markers are always interpreted as Ion Schema version markers and can never be valid open content.
+
+{% capture sample_code %}
+```ion
+penguins
+
+$ion_schema_2_0
+
+$note_to_self::"I really like penguins. Get ready!"
+3
+2
+1
+Go
+
+schema_header::{
+  _info: "This schema is about penguins."
+}
+
+(I <3 Penguins!)
+
+type::{
+  name: adelie,
+  type: penguin,
+  _region: antarctica,
+  _crested: false,
+  _banded: false,
+}
+
+"I have watched many penguin documentaries and know many facts about penguins."
+
+type::{
+  name: humboldt,
+  type: penguin,
+  _region: south_america,
+  _crested: false,
+  _banded: true,
+}
+
+FAQ::{
+  q: "What is the biggest type of penguin?",
+  a: "The emperor penguin."
```

Comment:
For accuracy, consider "biggest *extant* species of penguin". We don't want to anger fans of the late Colossus penguin.

Reply:
🐧 👍